### PR TITLE
[urdf] package.xml: add missing exec_depend to urdf_parser_plugin

### PR DESCRIPTION
**Pull request by tobiasneumann**
Because its been a build dependency that was probably not a visible problem in a normal ROS workspace.
However when used in context of openembedded (where the build space and runtime space are separated) this results in `urdf_parser_plugin` not been part of the image.

- FIX: [urdf] package.xml: add missing exec_depend to urdf_parser_plugin

fixup for ros2/urdf#13

### CODE REVIEW

**Review by clalancette — COMMENTED** [review]
So I don't totally see how this is possible.

`urdf_parser_plugin` is a header-only library; it only installs a parser.h file plus some CMake glue.

Further, this package (`urdf`) only uses `urdf_parser_plugin` in internal C++ files (in src/model.cpp and src/urdf_plugin.cpp).  So I don't see how this can require an `exec_depend`.

Can you give us some more information about what failed and why you think this is the fix?

**Comment by tobiasneumann** [comment]
Without it the exception `package 'urdf_parser_plugin' not found, searching: [/usr]` is thrown.
I don't know the origin, but i would presume that this occurs at the instantiation of `loader_`.

**Comment by clalancette** [comment]
> I don't know the origin, but i would presume that this occurs at the instantiation of `loader_`.

But the thing is that it is a templated class, so that code is all inlined at build time.  `urdf_parser_plugin` shouldn't be needed at runtime at all.  That's why I'm confused.

It smells to me like the problem is somewhere else.  But I'll defer to @sloretz here, who did the work to add in `urdf_parser_plugin`.

**Comment by tobiasneumann** [comment]
I created a little bit unconventional way to reproduces this with a normal ROS workspace.

- Clone and build https://github.com/ros/urdf_tutorial.git (branch `ros2`)
- Use the launch file from the tutorial with any urdf `ros2 launch urdf_tutorial display.launch.py model:=<any-urdf>` to check that this works.

---

- move or delete `urdf_parser_plugin` from wherever you have it installed.
- try to restart the launch file again, this will result in the exception been thrown.

**Comment by clalancette** [comment]
> I created a little bit unconventional way to reproduces this with a normal ROS workspace.

Ah, thank you!  That helps a lot.

And now I see what is going on.  During instantiation of the `urdf::Model` class, it instantiates a `urdf::ModelImplementation`, which in turn instantiates a `pluginlib::ClassLoader` [here](https://github.com/ros2/urdf/blob/633be51b8a1899f2939257f88ef7617327ffa625/urdf/src/model.cpp#L64).  Note that the package name passed in there is `urdf_parser_plugin`.  Down in the pluginlib [implementation](https://github.com/ros/pluginlib/blob/b0388a8b8aa50170bcfbbfbf6da1c54dad6532da/pluginlib/include/pluginlib/class_loader_imp.hpp#L91), you see that we call `ament_index_cpp::get_package_prefix` on that package name.  So if it doesn't exist, then we get an exception.

So you are totally correct; there *is* a runtime dependency here, it is just a somewhat unconventional one.  In that case, your patch is totally correct, but I'll request that you add a comment above the new `exec_depend` explaining the situation.  Once that is in, I'm happy to approve and run CI on it.

Thanks again.

**Comment by tobiasneumann** [comment]
I tried to summarize the main reason why its needed from your findings.
And hope that this is as you've expected it.

File: urdf/package.xml
Comment (clalancette) [inline urdf/package.xml]:
```suggestion
  <!-- while `urdf_parser_plugin` is a header only lib, `pluginlib` requires the package to exist during runtime. -->
```

**Review by clalancette — CHANGES_REQUESTED** [review]
Small grammatical fixes.  Once those are in, I'm happy to approve and run CI on it.  Thanks.

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17548/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12086/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18114/)

**Pull request merged**
